hack/update-template/unstructured-modification.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+// all of these have been pulled from k8s.io/apimachinery/pkg/apis/meta/v1/unstructured
+// this issue https://github.com/kubernetes/apimachinery/issues/138 explains why I needed to duplicate these funcs
```

Comment:
The issue doesn't really explain why you can't just import the package. The issue explains a problem `with` the package, but it doesn't explain what you've changed or what the package does different to the code in this file.

If this is the same code like in apimachinery... why do we copy it?

Reply:
yeah, I will write it there aswell. good point

Reply:
That is much better